capstone/src/main/webapp/scripts/auth.js

```diff
@@ -39,19 +39,57 @@ function createLoginUrl() {
     if (login.loggedIn) {
       afterUserSignsIn();
     }
- });
+  });
 }
 
 /*
- * Sign up the user for the app if they have not already signed up.
+ * Check if a user is new.
  * Redirect logged in users to the correct page.
  */
 function afterUserSignsIn() {
-  // Send a POST request to the servlet which checks if a user is newly registered.
-  fetch('/checkNewUser', {method: 'POST'})
-  .then(() => {
-    // TODO: Redirect newly registered users to a separate place.
-    // Redirect to group page once logged in.
+  // Send a request to the servlet which checks if a user is new to the webpage.
+  fetch('/checkNewUser')
+ .then(response => response.json())
+ .then((user) => {
+    if (user.isUserNew) {
+      openRegisterModal();
+    } else {
+      // Redirect to group page once logged in.
+      window.location.href = 'group.html';
+    }
+  });
+}
+
+/*
+ * Opens up a modal for new users to register.
+ */
+function openRegisterModal() {
+  let modal = document.getElementById('register-modal');
+  modal.classList.toggle('show-modal');
+}
+
+/*
+ * Sign up new user for the app.
+ */
+function createNewUser() {
+  const firstName = document.getElementById('first').value;
```

Comment:
for the required fields such as "First name *" are you checking that if the user doesn't type anything then they can't submit?

Reply:
Yes, in the HTML they're marked as required so if they're empty it prompts the user to fill them out

Reply:
I looked again and it looks like there actually is an intermediate state where it's creating a UserEntity, even if the form is invalid. Though after it prompts the user to fill out the fields and user submits again, it ends up being overwritten. We probably still don't want that, so I'm including a check just in case.